sql/core/src/main/scala/org/apache/spark/sql/functions.scala

```diff
@@ -2546,15 +2546,39 @@ object functions {
   def soundex(e: Column): Column = withExpr { SoundEx(e.expr) }
 
   /**
-   * Splits str around pattern (pattern is a regular expression).
+   * Splits str around matches of the given regex.
    *
-   * @note Pattern is a string representation of the regular expression.
+   * @param str a string expression to split
+   * @param regex a string representing a regular expression. The regex string should be
+   *              a Java regular expression.
    *
    * @group string_funcs
    * @since 1.5.0
    */
-  def split(str: Column, pattern: String): Column = withExpr {
-    StringSplit(str.expr, lit(pattern).expr)
+  def split(str: Column, regex: String): Column = withExpr {
```

Comment:
Shall we just keep it as `pattern`? I think we don't better change the name. Doesn;t `pattern` also make sense?

Reply:
The reason I changed it is that every time we mentioned `pattern` in the comments/docs, we always added a phrase like "pattern, which is a regular expression ..."

just felt like unnecessary explanation needed if we called the variable `regex`. Happy to change if you think necessary though!

Reply:
Yea, I don't think we should change the name in case either makes sense in a way.

Reply:
by having it as `regex`, the documentation will require less explanation. @HyukjinKwon if you are ok with keeping it as `regex` I think we should prefer to keep this change.

Happy to revert as well of course